srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/SrcdepsInitMojo.java

```diff
@@ -0,0 +1,516 @@
+                projectBuildingRequest.setProcessPlugins(false);
+                projectBuildingRequest.setRepositoryMerging(ProjectBuildingRequest.RepositoryMerging.REQUEST_DOMINANT);
+                projectBuildingRequest.setSystemProperties(session.getSystemProperties());
+                projectBuildingRequest.setRepositorySession(repoSession);
+
+                Artifact pomArtifact = repositorySystem.createProjectArtifact(g, a, v, "compile");
+                try {
+                    ProjectBuildingResult result = projectBuilder.build(pomArtifact, projectBuildingRequest);
+                    MavenProject dependencyProject = result.getProject();
+                    String url = getScmUrl(dependencyProject, null);
+                    if (url == null) {
+                        log.warn(String.format("No SCM connection for artifact [%s:%s:%s]", g, a, v));
+                    } else {
+                        url = url.substring("scm:".length());
+                        if (isScmUrlSupported(url)) {
+                            log.info(String.format("Found SCM URL [%s] for GAV [%s]", url, gav));
+                            this.add(url, gav);
+                        } else {
+                            log.debug(String.format("Unsupported SCM URL [%s] for GAV [%s]", url, gav));
+                        }
+                    }
+                } catch (ProjectBuildingException e) {
+                    final String msg = String.format("Could not resolve [%s] using remote repositories [%s]",
+                            pomArtifact, remoteRepos);
+                    if (failOnUnresolvable) {
+                        throw new MojoExecutionException(msg, e);
+                    } else {
+                        log.warn(msg);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Creates a new {@link ScmRepository.Builder} and sets its selectors, SCM url and id based on the information
+         * available in this {@link ScmRepositoryIndex}.
+         *
+         * @param url
+         *            the SCM URL to create a new {@link ScmRepository.Builder} for
+         * @return a new {@link ScmRepository.Builder}
+         */
+        public ScmRepository.Builder createRepository(String url) {
+
+            ScmRepository.Builder repoBuilder = ScmRepository.builder();
+            Set<String> selectors = new TreeSet<>();
+
+            Map<String, Integer> groupIdCntMap = urlGroupIdMap.get(url);
+            if (groupIdCntMap.size() == 1) {
+                String groupId = groupIdCntMap.keySet().iterator().next();
+                Set<String> urls = groupIdUrlMap.get(groupId);
+
+                if (urls.size() == 1) {
+                    repoBuilder.id(groupId);
+                    selectors.add(groupId);
+                } else {
+                    Set<Gav> gavs = urlGavMap.get(url);
```

Comment:
I'm lost here. What's the use case? When will it need to have different repositories for a given GAV?